nestkernel/synaptic_element.h

```diff
@@ -0,0 +1,297 @@
+/*
+ *  synaptic_element.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/**
+ * \file synaptic_element.h
+ * Definition of SynapticElement which is capable of
+ * managing a synaptic element of a node.
+ * \author Mikael Naveau
+ * \date July 2013
+ */
+
+#ifndef SYNAPTIC_ELEMENT_H
+#define SYNAPTIC_ELEMENT_H
+
+#include "dictdatum.h"
+#include "dictutils.h"
+#include "histentry.h"
+#include <cmath>
+
+namespace nest
+{
+
+class SynapticElement;
+class Archiving_Node;
+
+/**
+ * \class GrowthCurve
+ * Defines the way the number of synaptic elements changes through time
+ * according to the calcium concentration of the neuron.
+ */
+class GrowthCurve
```

Comment:
This class and some derived ones have private/protected member variables whose names should end in an underscore to be consistent with the conventions.